### PR TITLE
Update Makefile and swagger

### PR DESCRIPTION
**Pull request by ywk253100**

### CODE REVIEW

File: src/jobservice/replication/transfer.go [hunk 1]
```diff
@@ -460,6 +462,14 @@ func (m *ManifestPusher) enter() (string, error) {
 func newRepositoryClient(endpoint string, insecure bool, credential auth.Credential, repository, scopeType, scopeName string,
 	scopeActions ...string) (*registry.Repository, error) {
 
+	domain, err := config.DomainName()
```
Comment (reasonerjt) [inline src/jobservice/replication/transfer.go]: Didn't we call it hostname in ui process?

File: src/ui/api/repository.go [hunk 2]
```diff
@@ -442,6 +443,15 @@ func newRepositoryClient(endpoint string, insecure bool, username, password, rep
 	scopeActions ...string) (*registry.Repository, error) {
 
 	credential := auth.NewBasicAuthCredential(username, password)
+
+	domain, err := config.DomainName()
+	if err != nil {
+		return nil, err
+	}
+	if err := os.Setenv("DOMAIN_NAME", domain); err != nil {
```
Comment (reasonerjt) [inline src/ui/api/repository.go]: Why set env? using env as global var?

File: src/jobservice/replication/transfer.go (hunk 1 above)
Comment (ywk253100) [inline src/jobservice/replication/transfer.go]: I think "hostname" is inaccurate as the value is the external URL which Harbor can be reached, the hostname is not the correct value if Harbor is deployed behind a proxy. So I change it to domain name instead.

File: src/common/utils/registry/auth/tokenauthorizer.go [hunk 3]
```diff
@@ -133,19 +133,24 @@ func (t *tokenAuthorizer) updateCachedToken(token string, expiresIn int) {
...
-	credential Credential
+	client               *http.Client
+	credential           Credential
+	tokenServiceEndpoint string
 }
 
 // NewStandardTokenAuthorizer returns a standard token authorizer. The authorizer will request a token
 // from token server and add it to the origin request
-func NewStandardTokenAuthorizer(credential Credential, insecure bool, scopeType, scopeName string, scopeActions ...string) Authorizer {
+// If tokenServiceURL is set, the token request will be sent to it instead of the server get from authorizer
```
Comment (reasonerjt) [inline src/common/utils/registry/auth/tokenauthorizer.go]: in the parm it's tokenServiceEndpoint

File: src/jobservice/replication/transfer.go [hunk 4]
```diff
@@ -457,10 +459,20 @@ func (m *ManifestPusher) enter() (string, error) {
...
+func newRepositoryClient(endpoint string, insecure bool, credential auth.Credential,
+	tokenServiceEndpoint, repository, scopeType, scopeName string,
 	scopeActions ...string) (*registry.Repository, error) {
 
-	authorizer := auth.NewStandardTokenAuthorizer(credential, insecure, scopeType, scopeName, scopeActions...)
+	domain, err := config.ExtEndpoint()
+	if err != nil {
+		return nil, err
+	}
+	if err := os.Setenv("DOMAIN_NAME", domain); err != nil {
```
Comment (reasonerjt) [inline src/jobservice/replication/transfer.go]: Why still Setenv?

**Pull request merged**